src/ru/nsu/ccfit/zuev/osu/game/GameScene.java

```diff
@@ -2075,18 +2086,22 @@ public void onCircleHit(int id, final float acc, final PointF pos,
             short sacc = (short) (acc * 1000);
             replay.addObjectResult(id, sacc, null);
         }
-        if(GameHelper.isFlashLight()){
 
+        if (cursorSprites != null) {
+            cursorSprites[0].updatePosIfAuto(pos.x, pos.y, activeObjects);
+        }
```

Comment:
Putting this in `onCircleHit` means the position of the cursor sprite will be updated after the circle is hit, which is correct. However, there is a drag animation towards the circle, which should be started right after the last object is hit or it has ended (slider/spinner). In this case, the animation is started upon hitting a circle (which is when the animation is supposed to end).

Additionally, this means that cursor movement towards a slider or a spinner will have no drag animation, which is inconsistent.